test/test_blobstore.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+
+	c "github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-check"
+)
+
+type BlobstoreSuite struct {
+	Helper
+}
+
+var _ = c.Suite(&BlobstoreSuite{})
+
+func (s *BlobstoreSuite) TestBlobstoreBackendSwitching(t *c.C) {
+	s3Config := os.Getenv("BLOBSTORE_S3_CONFIG")
+	if s3Config == "" {
+		// BLOBSTORE_S3_CONFIG should be set to a valid configuration like:
+		// backend=s3 access_key_id=xxx secret_access_key=xxx bucket=blobstore-ci region=us-east-1
+		t.Skip("missing BLOBSTORE_S3_CONFIG env var")
+	}
+
+	r := s.newGitRepo(t, "http")
+	t.Assert(r.flynn("create", "blobstore-backend-test"), Succeeds)
+	t.Assert(r.git("push", "flynn", "master"), Succeeds)
+
+	// set default backend to S3 without printing secrets
+	cmd := exec.Command("sh", "-c", args.CLI+` -a blobstore env set "BACKEND_S3=$BLOBSTORE_S3_CONFIG" DEFAULT_BACKEND=s3`)
+	cmd.Env = flynnEnv(flynnrc)
+	cmd.Env["BLOBSTORE_S3_CONFIG"] = s3Config
```

Comment:
`flynnEnv` is a copy of `os.Environ()` just with `FLYNNRC` set, so it will already have `BLOBSTORE_S3_CONFIG`